py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,339 @@
+                try:
+                    status = modem_proxy.GetStatus()
+                except:
+                    # if no modem is selected, try all available
+                    continue
+
+                # start to build return data dict
+                try:
+                    data['signal'] = status['signal-quality'][0]
+                    self.py3.threshold_get_color(data['signal'], 'signal')
+                except:
+                    data['signal'] = STRING_UNKNOWN
+
+                try:
+                    highest_access_bit = 1 << (
+                        status['access-technologies'].bit_length() - 1)
+                    data['netgen'] = self.speed[highest_access_bit]
+                    self.py3.threshold_get_color(highest_access_bit, 'netgen')
+                except:
+                    data['netgen'] = STRING_UNKNOWN
+
+                # if registred on network, get operator name
+                try:
+                    if self.rstates[status['m3gpp-registration-state']] in (
+                            'HOME', 'ROAMING'):
+                        data['operator'] = status['m3gpp-operator-name']
+                    else:
+                        data['operator'] = STRING_UNKNOWN
+                except:
+                    data['operator'] = STRING_UNKNOWN
+
+                # use human readable format
+                data['status'] = self.states[status['state']]
+
+                # get status color
+                self.py3.threshold_get_color(status['state'], 'status')
+                self.py3.threshold_get_color(status['state'], 'color')
+
+                # TODO: maybe change to a rstates condition?
+                if status['state'] >= 0 and status['state'] <= 7:  # no service
+                    full_text = self.py3.safe_format(self.format_no_service,
+                                                     data)
+                elif status['state'] == 11:  # service
+                    bearer = modem_proxy.Bearers[0]
+
+                    # get nic name
+                    interface = self._get_interface(bearer)
+                    data['nic'] = interface
+
+                    # Get network config
+                    network_config = self._get_network_config(bearer)
+                    ipv4 = network_config.get('ipv4', {})
+                    ipv6 = network_config.get('ipv6', {})
+
+                    # Add network config to data dict
+                    data['format_ipv4'] = self.py3.safe_format(
+                        self.format_ipv4, ipv4)
+                    data['format_ipv6'] = self.py3.safe_format(
+                        self.format_ipv6, ipv6)
+
+                    if ipv4 or ipv6:
+                        color = self.py3.COLOR_GOOD
+
+                        # Get connection statistics
+                        stats = self._get_stats(bearer)
+                        data['format_stats'] = self.py3.safe_format(
+                            self.format_stats, stats)
```

Comment:
It seems to me that checking `if ipv4 or ipv6` might be useless as we know the modem is connected. Try getting the `stats` anyway... At most, we get back nothing. This requires the color `state` threshold to be working. (We want to have one `format` everywhere eventually).